AppinfoJsonFile.js

```diff
@@ -0,0 +1,309 @@
+        } catch (e) {
+            logger.warn("Could not read file: " + p);
+        }
+    }
+};
+
+/**
+ * Return the set of resources found in the current JavaScript file.
+ *
+ * @returns {TranslationSet} The set of resources found in the
+ * current JavaScript file.
+ */
+AppinfoJsonFile.prototype.getTranslationSet = function() {
+    return this.set;
+}
+
+// we don't localize or write appinfo.json source files
+AppinfoJsonFile.prototype.write = function() {};
+
+/**
+ * Return the location on disk where the version of this file localized
+ * for the given locale should be written.
+ * @param {String] locale the locale spec for the target locale
+ * @returns {String} the localized path name
+ */
+AppinfoJsonFile.prototype.getLocalizedPath = function(locale) {
+    var rootPath = this.project.getResourceDirs("json")[0] || ".";
+    var fullPath = "";
+
+    var splitLocale = locale.split("-");
+    if (this.baseLocale) {
+        fullPath = "/" + splitLocale[0];
+    } else {
+        for (var i=0; i < splitLocale.length; i++) {
+            fullPath += "/"+ splitLocale[i];
+        }
+    }
+    return rootPath + fullPath;
+};
+
+/**
+ * Localize the text of the current file to the given locale and return
+ * the results.
+ *
+ * @param {TranslationSet} translations the current set of translations
+ * @param {String} locale the locale to translate to
+ * @returns {String} the localized text of this file
+ */
+AppinfoJsonFile.prototype.localizeText = function(translations, locale) {
+    var output = {};
+    for (var property in this.parsedData) {
+        if (this.schema[property]){
+            var text = this.parsedData[property];
+            var key = this.makeKey(this.API.utils.escapeInvalidChars(text));
+            var tester = this.API.newResource({
+                resType: "string",
+                project: this.project.getProjectId(),
+                sourceLocale: this.project.getSourceLocale(),
+                reskey: key,
+                datatype: this.datatype || "javascript"
+            });
+            var hashkey = tester.hashKeyForTranslation(locale);
+            var translated = translations.getClean(hashkey);
+
+            if (translated) {
+                output[property] = translated.target;
+            } else {
+                logger.trace("New string found: " + text);
+                var r = this.API.newResource({
+                    resType: "string",
+                    project: this.project.getProjectId(),
+                    sourceLocale: this.project.getSourceLocale(),
+                    source: this.API.utils.escapeInvalidChars(text),
+                    targetLocale: locale,
+                    target: this.API.utils.escapeInvalidChars(text),
+                    reskey: key,
+                    state: "new",
+                    datatype: this.datatype || "javascript"
+                });
+                this.type.newres.add(r);
+            }
+        }
+    }
+    return output;
```

Comment:
This is supposed to return a string, but it returns an object.

Reply:
I fixed it.